contracts/Governance/GovernorBravoDelegate.sol

```diff
@@ -156,8 +157,18 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV1, GovernorBravoE
         require(state(proposalId) != ProposalState.Executed, "GovernorBravo::cancel: cannot cancel executed proposal");
 
         Proposal storage proposal = proposals[proposalId];
-        require(msg.sender == proposal.proposer || comp.getPriorVotes(proposal.proposer, sub256(block.number, 1)) < proposalThreshold, "GovernorBravo::cancel: proposer above threshold");
 
+        // Proposer can cancel
+        if(msg.sender != proposal.proposer) {
+            // Whitelisted proposers can't be canceled for falling below proposal threshold
+            if(isWhitelisted(proposal.proposer)) {
+                require((comp.getPriorVotes(proposal.proposer, sub256(block.number, 1)) < proposalThreshold) && msg.sender == whitelistGuardian, "GovernorBravo::cancel: whitelisted proposer");
+            }
+            else {
+                require((comp.getPriorVotes(proposal.proposer, sub256(block.number, 1)) < proposalThreshold), "GovernorBravo::cancel: proposer above threshold");
+            }
```

Comment:
```suggestion
            require((comp.getPriorVotes(proposal.proposer, sub256(block.number, 1)) < proposalThreshold), "GovernorBravo::cancel: proposer above threshold");
            // Only whitelist guardian can cancel whitelisted proposers falling below proposal threshold
            if(isWhitelisted(proposal.proposer)) {
                require(msg.sender == whitelistGuardian, "GovernorBravo::cancel: whitelisted proposer");
            }
```

I think this might be slightly cleaner and also outputs the correct error message for all cases. 

The current implementation will output `"GovernorBravo::cancel: whitelisted proposer"` instead of `"GovernorBravo::cancel: proposer above threshold"` even when `msg.sender == whitelistGuardian` but `comp.getPriorVotes(proposal.proposer, sub256(block.number, 1)) >= proposalThreshold`.

Reply:
+1

Reply:
Yes, definitely better form. A bit too late to implement now though